includes/adapter/ShippingRecipientFromWCOrder.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\WooCommerce\Includes\Adapter;
+
+use MyParcelNL\Sdk\src\Helper\ValidateStreet;
+use MyParcelNL\Sdk\src\Model\Fulfilment\OrderLine;
+use MyParcelNL\Sdk\src\Model\Recipient;
+use WC_Order;
+use WC_Order_Item;
+use WPO\WC\MyParcel\Compatibility\Order as WCX_Order;
+
+class ShippingRecipientFromWCOrder extends RecipientFromWCOrder
+{
+    /**
+     * @var WC_Order
+     */
+    private $order;
+
+    /**
+     * @var string
+     */
+    protected $carrierCountry;
+
+    /**
+     * RecipientFromOrder constructor.
+     *
+     * @param WC_Order $order
+     * @param string
+     *
+     * @throws \Exception
+     */
+    public function __construct(WC_Order $order, string $carrierCountry)
+    {
+        $this->order = $order;
+        $this->carrierCountry = $carrierCountry;
+        parent::__construct();
+    }
+
+    /**
+     * @return array
+     */
+    public function prepareOrderData(): array
+    {
+        $addressParts = $this->getAddressDetails();
+
+        $fullStreet           = $addressParts['full_street'];
+        $cc                   = $this->order->get_shipping_country();
+        $city                 = $this->order->get_shipping_city();
+        $person               = $this->getShippingName();
+        $company              = $this->order->get_shipping_company();
+        $email                = $this->getEmailAddress($this->order);
+        $phone                = $this->getPhoneNumber($this->order);
+        $postalCode           = $this->order->get_shipping_postcode();
+        $region               = $this->order->get_shipping_state();
+        $streetAdditionalInfo = $addressParts['street_additional_info'];
+
+        return [
```

Comment:
Er is nu best wel veel dubbele code, ik denk dat de structuur aangepast moet worden